.ci/prepare.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import print_function
+
+
+import hashlib
+import sys
+import requests
+import os
+from shell import ex
+
+JOB_LIST_DIR = '.ci/jobs'
+
+
+def find_changed_dockerfiles():
```

Comment:
It would be great to add some documentation here. It is a little difficult to follow what is happening at each step here.